src/trading/ShareToken.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/token/VariableSupplyToken.sol';
+import 'ROOT/reporting/Market.sol';
+
+
+contract ShareToken is DelegationTarget, VariableSupplyToken, Typed, Initializable {
+
+    //FIXME: Delegated contracts cannot currently use string values, so we will need to find a workaround if this hasn't been fixed before we release
+    string constant public name = "Shares";
+    uint256 constant public decimals = 18;
+    string constant public symbol = "SHARE";
+
+    Market private market;
+    uint8 private outcome;
+
+    function initialize(Market _market, uint8 _outcome) external beforeInitialized returns(bool) {
+        endInitialization();
+        market = _market;
+        outcome = _outcome;
+    }
+
+    function createShares(address _owner, uint256 _fxpValue) onlyWhitelistedCallers external returns(bool) {
+        mint(_owner, _fxpValue);
+        return true;
+    }
+
+    function destroyShares(address _owner, uint256 _fxpValue) onlyWhitelistedCallers external returns(bool) {
```

Comment:
I think at some point we should rename this to `burn` and the other to `mint` and then have them apply the modifier and call `super.mint()`/`super.burn()` (or whatever the syntax for that is).  That is probably better as a separate PR, possibly after we have finished the Solidity migration.